lib/ipmi_backend_utils.pm

```diff
@@ -528,4 +528,22 @@ sub set_grub_terminal_and_timeout {
     }
 }
 
+sub reconnect_when_ssh_console_broken {
+
+    # Switch to sol console to check serial console output
+    # It is useful in the case of host crash and reboot
+    record_info("WARN", "ssh connection is broken and switch to SOL console", result => 'fail');
+    select_console 'sol', await_console => 0;
+    save_screenshot;
+    wait_still_screen 5;
+    # Wait host bootup if it crashes
+    check_screen("linux-login", 120);
+    reset_consoles;
+    record_info("switch back to ssh console to collect logs");
+    select_console('root-ssh');
```

Comment:
What if ```select_console('root-ssh');``` fails here ? Did you verify this scenario ?

Reply:
Going to have a try. thanks.